libsel4/include/sel4/macros.h

```diff
@@ -19,7 +19,14 @@
 #define SEL4_PACKED             __attribute__((packed))
 #define SEL4_DEPRECATED(x)      __attribute__((deprecated(x)))
 #define SEL4_DEPRECATE_MACRO(x) _Pragma("deprecated") x
+
+
+#ifdef CONFIG_VERIFICATION_BUILD
+/* __builtin_offsetof() is currently not supported by the C Parser */
+#define SEL4_OFFSETOF(type, member) ((word_t)&(((type *)0)->member))
+#else /* not CONFIG_VERIFICATION_BUILD */
 #define SEL4_OFFSETOF(type, member) __builtin_offsetof(type, member)
```

Comment:
If the new one is proven to be correct, why also have the built-in one?

Reply:
Actually, the macro (and various others) violates the C specs, see explanation at https://en.wikipedia.org/wiki/Offsetof. That is why using `__builtin_offsetof()` is the best option and ideally support for this in the verification toolchain should be added. I agree that the current solution is not really optimal and not using `__builtin_offsetof()` might be something to consider. Opinions from others?

Reply:
I disagree that it violates the C specs as it is doing pointer arithmetic and not actually following the pointer, and the arithmetic itself has no side effects, so could never have undefined behaviour. But I have a practical view of standards.